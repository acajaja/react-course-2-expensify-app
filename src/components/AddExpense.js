import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <section>
                    <h1>Add New Expense</h1>
                    <ExpenseForm
                        onSubmit={this.onSubmit} />
                </section>
            </div>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpense);
