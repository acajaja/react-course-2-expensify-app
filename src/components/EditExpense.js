import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpense extends React.Component {
    onClick = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    onSubmit = (updates) => {
        this.props.startEditExpense(this.props.expense.id, updates);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <section>
                    <div className="page-header">
                        <div className="content-container">
                            <h1 className="page-header__title">Edit Expense</h1>
                        </div>
                    </div>
                    <div className="content-container">
                        <ExpenseForm
                            expense={this.props.expense}
                            onSubmit={this.onSubmit}
                        />
                        <div>
                            <button
                                className="button button--remove"
                                onClick={this.onClick}
                                title="Remove this expense"
                            >Remove Expense</button>
                        </div>
                    </div>
                </section>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, updates) => dispatch(startEditExpense(id, updates)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
