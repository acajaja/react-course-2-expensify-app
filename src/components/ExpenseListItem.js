import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <section>
            <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
            <p>
                {numeral(amount / 100).format('$0,0.00')}
                -
                created at: {moment(createdAt).format('MMMM Do, YYYY')}
            </p>
        </section>
    </div>
);

export default ExpenseListItem;
