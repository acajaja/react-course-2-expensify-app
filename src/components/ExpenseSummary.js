import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({expensesCount, totalAmount}) => {
    return (
        <section className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expensesCount}</span> expenses
                    totaling <span>{numeral(totalAmount / 100).format('$0,0.00')}</span>.
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </section>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length,
        totalAmount: expensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
