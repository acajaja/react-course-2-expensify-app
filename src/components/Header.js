import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

// Define for testing
export const Header = ({ startLogout }) => (
    <header className="header">
        <nav>
            <div className="content-container">
                <div className="header__content">
                    <Link className="header__title" to="/dashboard" title="Dashboard">
                        <h1>Expensify</h1>
                    </Link>
                    <button className="button button--login" onClick={(e) => {
                        e.preventDefault();
                        startLogout();
                    }} title="Logout">Logout</button>
                </div>
            </div>
        </nav>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
