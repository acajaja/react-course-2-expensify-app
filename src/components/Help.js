import React from 'react';

const HelpPage = () => (
    <div>
        <p>HelpPage component!</p>
    </div>
);

export default HelpPage;
