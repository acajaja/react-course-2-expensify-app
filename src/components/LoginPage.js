import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>Yo! Get your expenses together!</p>
            <button className="button" onClick={(e) => {
                e.preventDefault();
                startLogin();
            }} title="Login">Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
