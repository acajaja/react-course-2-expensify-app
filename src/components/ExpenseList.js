import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <section>
            <div className="list-header">
                <div className="show-for-mobile">Expenses</div>
                <div className="show-for-desktop">Expense</div>
                <div className="show-for-desktop">Amount</div>
            </div>
            <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item--message">
                        <p>No expenses</p>
                    </div>
                ) : (
                    props.expenses.map((expense) => (
                        <ExpenseListItem key={expense.id} {...expense} />
                    ))
                )
            }
            </div>
        </section>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
