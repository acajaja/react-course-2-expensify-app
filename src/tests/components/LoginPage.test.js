import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('Test render default login page', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('Test login button click', () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={onClickSpy} />);
    wrapper.find('button').simulate('click', {
        preventDefault: () => {}
    });
    expect(wrapper).toMatchSnapshot();
    expect(onClickSpy).toHaveBeenCalled();
});
