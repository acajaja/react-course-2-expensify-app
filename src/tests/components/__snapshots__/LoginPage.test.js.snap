// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test login button click 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      Yo! Get your expenses together!
    </p>
    <button
      className="button"
      onClick={[Function]}
      title="Login"
    >
      Login with Google
    </button>
  </div>
</div>
`;

exports[`Test render default login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      Yo! Get your expenses together!
    </p>
    <button
      className="button"
      onClick={[Function]}
      title="Login"
    >
      Login with Google
    </button>
  </div>
</div>
`;
